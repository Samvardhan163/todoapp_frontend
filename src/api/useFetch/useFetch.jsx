import React from "react";
import axios from "axios";
import { useState, useEffect } from "react";
export default function useFetch(url) {
  const [data, setdata] = useState(null);
  const [isPending, setPending] = useState(true);
  const [error, seterror] = useState(null);

  const fetchTodos = () => {
    axios
      .get(url)
      .then((res) => {
        if (res.status !== 200) {
          throw Error("Data is not fetched");
        }
        return res.data;
      })
      .then((data) => {
        setdata(data);
        setPending(false);
        seterror(null);
      })
      .catch((err) => {
        if (err.name === "AbortError") {
          console.log("fetch aborted");
        } else {
          setPending(false);
          seterror(err.message);
        }
      });
  };

  useEffect(() => {
    fetchTodos();
  }, [url]);

  const refetch = () => {
    fetchTodos();
  };

  return { data,error,refetch };
}
