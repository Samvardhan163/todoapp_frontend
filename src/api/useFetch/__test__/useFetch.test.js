import { renderHook } from "@testing-library/react-hooks";
import useFetch from "../useFetch";

describe("useFetch", () => {
  it("should exist", () => {
    const { result } = renderHook(() => useFetch());
    expect(result.current).toBeDefined();
  });
  it("should return the data from the server", async () => {
    const { result, waitForNextUpdate } = renderHook(() =>
      useFetch("http://localhost:8080/api/todo")
    );
    await waitForNextUpdate();
    expect(result.current.data).toMatchSnapshot();
  });
});
