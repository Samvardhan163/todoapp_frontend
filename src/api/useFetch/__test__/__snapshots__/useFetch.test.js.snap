// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useFetch should return the data from the server 1`] = `
Array [
  Object {
    "completed": "true",
    "description": "playing",
    "id": 1,
    "priority": "true",
  },
]
`;
